### PR TITLE
feat: theming frontend

### PR DESCRIPTION
**Pull request by mindspank**
INSERT DESCRIPTION HERE

**Checklist:**
- [ ] Covered by tests
- [ ] Ran it and it works as intended
- [ ] Reviewed the diff before requesting a review
- [ ] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [ ] I'm proud of this work!

### CODE REVIEW

File: web-common/src/features/themes/css-sanitizer.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
+/**
+ * CSS Sanitizer for Theme System
+ * 
+ * Extracts only known safe CSS variables from user-provided CSS
+ * to prevent XSS attacks through CSS injection.
+ */
+
+// Allowed CSS variables to prevent XSS attacks
+// Only these variables can be set through custom CSS
+export const ALLOWED_CSS_VARIABLES = new Set([
```
Comment (mindspank) [inline web-common/src/features/themes/css-sanitizer.ts]: For now a large list of allowed variables, needs to be reduced for release
